src/verifier/builder_visitor.rs

```diff
@@ -478,31 +481,61 @@ impl WhammVisitorMut<()> for SymbolTableBuilder<'_> {
 
         script.fns.iter_mut().for_each(|f| self.visit_fn(f));
         script.global_stmts.iter_mut().for_each(|stmt| {
-            if let Statement::Decl { ty, var_id, .. } = stmt {
-                if let Expr::VarId { name, .. } = &var_id {
-                    // Add global variable to script globals (triggers the init_generator to emit them!)
-                    script.globals.insert(
-                        name.clone(),
-                        Global {
-                            def: Definition::User,
-                            ty: ty.clone(),
-                            var_name: var_id.clone(),
-                            value: None,
-                        },
-                    );
-                } else {
-                    self.err.unexpected_error(
-                        true,
-                        Some(format!(
-                            "{} \
-                Variable declaration var_id is not the correct Expr variant!!",
-                            UNEXPECTED_ERR_MSG
-                        )),
-                        None,
-                    );
+            match stmt {
```

Comment:
This could be simplified quite a bit (the cases are the same except for `report` being true/false)